src/main/java/ch/njol/skript/expressions/ExprFishingState.java

```diff
@@ -0,0 +1,52 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerFishEvent.State;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.EventValueExpression;
+import ch.njol.skript.lang.ExpressionType;
+
+@Name("Fishing State")
+@Description("The <a href='/classes.html#fishingstate'>fishing state</a> of a fishing event.")
+@Examples("fishing state is failed or in ground")
+@Since("INSERT VERSION")
+public class ExprFishingState extends EventValueExpression<State> {
+	
+	static {
+		Skript.registerExpression(ExprFishingState.class, State.class, ExpressionType.SIMPLE, "[the] [event-]fish[ing]( |-)state");
```

Comment:
I'm against having `[event-]` as the point of having `[event-]` is for classinfos, and a fish is not a classinfo. At least at this current time.

event- signifys a type from the event. It shouldn't be used to cover up conflicts. Especially if a type was to actually be added with that name, it will have repercussions down the road and is bad design for add-on developers, as Skript is first to register.

Reply:
Sure thing, anyone else having an opinion about this? would it actually make any parsing priority problems to use `[event-]` in expressions?

Reply:
@TheLimeGlass the expression is fish state, not fish, and fishing state is a Skript type